lib/web_ui/lib/src/engine/surface/scene_builder.dart

```diff
@@ -157,6 +157,23 @@ class SurfaceSceneBuilder implements ui.SceneBuilder {
     throw UnimplementedError();
   }
 
+  /// Pushes an image filter operation onto the operation stack.
+  ///
+  /// The given filter is applied to the children's rasterization before compositing them into
+  /// the scene.
+  ///
+  /// {@macro dart.ui.sceneBuilder.oldLayer}
+  ///
+  /// {@macro dart.ui.sceneBuilder.oldLayerVsRetained}
+  ///
+  /// See [pop] for details about the operation stack.
+  @override
+  ui.ImageFilterEngineLayer pushImageFilter(ui.ImageFilter filter,
+                                            {ui.ImageFilterEngineLayer oldLayer}) {
+    assert(filter != null);
+    throw UnimplementedError();
```

Comment:
please implement this before landing

Reply:
(web is a first class part of our offering so anything that lands on engine must also land on web and vice versa, unless it's impossible to implement on one or the other, in which case we need to think about it more carefully)

Reply:
This is modeled on pushColorFilter which has no implementation on web. Have the standards changed since then?

Yegor reviewed this PR and told me to file an issue, which I did.

pushColorFilter issue: https://github.com/flutter/flutter/issues/40649
pushImageFilter issue: https://github.com/flutter/flutter/issues/47163

Reply:
I had a discussion with @yjbanov and we agreed that we need a formal process moving forward to deal with implementing new features in both the C++ and web versions of the engine. For the short term it would seem that having coordinated tasks for the 2 teams to implement the feature in both places is the most pragmatic process.

I originally created the issue linked above on his suggestion, but we also need a formal process to coordinate buy-in for implementation since the tasks will typically involve work from members of 2 different teams. He is on board with the importance of this particular feature and will pursue in the next day or two to get buy-in for the work that needs to be done for it on web.

Note that the web still doesn't have a blur implementation that works in all environments for BackdropFilter and so there is a workaround that produces a limited visual effect for that widget on web. But we can and should at least get to the point where `ImageFiltered(bgwidget, filter)` produces the same visual effect as `Stack[bgwidget, BackdropFilter]` (a common use of BackdropFilter mentioned by developers).

Reply:
Yeah, we still need to find a good process for implementing new functionality in the engine synchronously on all platforms. I think for now, notifying the Web team and leaving a TODO is OK. We certainly should not start using the new functionality in the framework before we have the Web implementation.

Reply:
I have the ImageFiltered widget ready to go, but I can hold that off until we can implement it in web.

Reply:
Yes, please wait until we implement it.

Reply:
@yjbanov is close to a web implementation so I will consider this comment stream resolved...